Unity-C#/README.md

```diff
@@ -0,0 +1,263 @@
+#Preamble 
+The below style uses the MSDN Style guidelines as a base to work with Unity, however because some of the nuances in Unity this style adapts and breaks from some of the standard conventions to obtain a higher level of readability.
+
+MSDN Style: http://msdn.microsoft.com/en-us/library/ff926074.aspx
+
+MSDN Class Styles documenting that public fields are not supposed to be used: http://msdn.microsoft.com/en-us/library/ms229012(v=vs.110).aspx
+
+MSDN XML Comments: http://msdn.microsoft.com/en-us/library/b2s063f7.aspx
+
+There is also a Unity style guide however because they have used Properties, which is not supported in the Unity Editor, and the style is almost exactly the same as the MSDN style, the MSDN style was used where possible.
+Unity Style: http://wiki.unity3d.com/index.php/Csharp_Coding_Guidelines
+
+# Style
+
+## Indentation
+Yes, a highly contentious topic so let's not waste time, at Wizcorp we use **tabs** for our projects. Tabs should be used for indentation, meaning at the beginning of lines only.
+
+
+## Line length
+Limit your lines to **120 characters**. We are not living in the 90's any more our screen got a lot bigger, but at the same time our brains are the same, so let's not go too far, readability come first.
+
+
+## Whitespace
+
+* Conventional operators SHOULD be surrounded by a space (including ternary operators).
+
+* Commas SHOULD be followed by a space.
+
+* Colons SHOULD be followed by a space.
+
+* Semi-colons in for statements SHOULD be followed by a space.
+
+* Semi-colons SHOULD NOT be preceded by a space.
+
+* Function calls and method calls SHOULD NOT be followed by a space. Example: `DoSomething(string name); // NOT DoSomething (string name)`
+
+* Logical units within a block SHOULD be separated by one blank line.
+
+* Curly braces SHOULD have inner spaces.
+
+
+## Braces
+In C# where the braces are placed is up to the user, however the C# MSDN standard is on a new line.  That said as long as the project has agreed to use on the same line there should be no complaints.
```

Comment:
I think there are for's and against's, I have sourced most of these and came to the conclusion that Unity developers were not thinking of styles of the other languages when they wrote the engine.

I would read the Unity style guide (which basically follows the MSDN guidelines)

I would work with the engine and have a look at the styles (or lack there of) from the engine code itself.

I have considered JS developers as it is OK to use one or the other, but it should depend on project, one project shouldn't have both (TJ IS THE WORST EXAMPLE OF THIS)

I have considered what C# is and where is came from rather than just JS developers.  For new hires that are Unity developers and not JS developers the C# standard is MSDN (for the most part) because C# is Microsofts language, so it makes sense when you start C# you tend to start with MSDN.

The trade off is against the C# standard / Unity standard, or JS standard.  Each is a different language, JS has the brackets on the same line because it HAS to, C# this restriction is not there.